pkg/services/pluginsintegration/pluginassets/pluginassets.go

```diff
@@ -69,6 +82,87 @@ func (s *Service) LoadingStrategy(_ context.Context, p pluginstore.Plugin) plugi
 	return plugins.LoadingStrategyFetch
 }
 
+// ModuleHash returns the module.js SHA256 hash for a plugin in the format expected by the browser for SRI checks.
+// The module hash is read from the plugin's MANIFEST.txt file.
+// The plugin can also be a nested plugin.
+// If the plugin is unsigned, an empty string is returned.
+// The results are cached to avoid repeated reads from the MANIFEST.txt file.
+func (s *Service) ModuleHash(ctx context.Context, p pluginstore.Plugin) string {
+	cachedValue, ok := s.moduleHashCache.Load(p.ID)
+	if ok {
+		return cachedValue.(string)
+	}
+	mh, err := s.moduleHash(ctx, p, "")
+	if err != nil {
+		s.log.Error("Failed to calculate module hash", "plugin", p.ID, "error", err)
+	}
+	s.moduleHashCache.Store(p.ID, mh)
+	return mh
+}
+
+// moduleHash is the underlying function for ModuleHash. See its documentation for more information.
+// It will read the module hash from the MANIFEST.txt in the [[plugins.FS]] of the provided plugin.
+// If childFSBase is provided, the function will try to get the hash from MANIFEST.txt for the provided children's
+// module.js file, rather than for the provided plugin.
+func (s *Service) moduleHash(ctx context.Context, p pluginstore.Plugin, childFSBase string) (r string, err error) {
+	// Ignore unsigned plugins
+	if !p.Signature.IsValid() {
+		return "", nil
+	}
+
+	// Always calculate the hash for CDN plugins.
+	// Do not calculate the hash for filesystem plugins, unless the corresponding feature toggle is enabled.
+	if !s.cdnEnabled(p.ID, p.Class) && !s.cfg.Features.FilesystemSriChecksEnabled {
+		return "", nil
+	}
+
+	if p.Parent != nil {
```

Comment:
If you wanted to make life a bit easier for yourself here, you could turn `p.Parent`'s type into pluginstore.Plugin so we don't have to go fetch it manually.